src/engraving/libmscore/undo.cpp

```diff
@@ -2083,6 +2083,28 @@ void InsertRemoveMeasures::removeMeasures()
     Fraction tick1 = fm->tick();
     Fraction tick2 = lm->endTick();
 
+    // remove beams from chordrests in affected area, they will be rebuilt later but we need
```

Comment:
Beams are kinda a special case here. The issue was caused by beams saving a list of their chordrests, but when a chordrest was deleted (i.e. still exist but are not part of the current score) those chordrests were not being removed from the beam. For example:
![image](https://user-images.githubusercontent.com/89263931/227242500-cb490713-d971-4462-a8c4-51e6d8fb568f.png)
If the highlighted measure has a timesig added after it, the measure (and all following) are actually deleted and all of the elements are cloned. The beam itself is not cloned, because it's still a valid non-deleted beam. This results in a beam with only 4 notes, but a list of 6 (including the phantom notes that have been deleted), which messes with the beam layout system.

When beams are laid out, they will be recomputed, which will only happen on first layout after the timesig is added.